chpl/chpl-service/src/main/java/gov/healthit/chpl/dao/impl/UserTestingLabMapDAOImpl.java

```diff
@@ -0,0 +1,171 @@
+        for (UserTestingLabMapEntity entity : result) {
+            dtos.add(new UserTestingLabMapDTO(entity));
+        }
+        return dtos;
+    }
+
+    @Override
+    public UserTestingLabMapDTO getById(Long id) {
+        Query query = entityManager.createQuery(
+                "from UserTestingLabMapEntity utlm " + "join fetch utlm.testingLab TestingLabEntity "
+                        + "join fetch utlm.user UserEntity" + "where (utlm.deleted != true) AND (utlm.id = :id) ",
+                UserTestingLabMapEntity.class);
+        query.setParameter("id", id);
+        List<UserTestingLabMapEntity> result = query.getResultList();
+
+        if (result.size() == 0) {
+            return null;
+        }
+        return new UserTestingLabMapDTO(result.get(0));
+    }
+
+    private UserTestingLabMapEntity create(UserTestingLabMapEntity entity) {
+        entity.setLastModifiedUser(getUserId(User.SYSTEM_USER_ID));
+        if (entity.getRetired() == null) {
+            entity.setRetired(false);
+        }
+        if (entity.getDeleted() == null) {
+            entity.setDeleted(false);
+        }
+        entityManager.persist(entity);
+        return entity;
+    }
+
+    private UserTestingLabMapEntity update(UserTestingLabMapEntity entity) {
+        entity.setLastModifiedUser(getUserId(User.SYSTEM_USER_ID));
+        entityManager.persist(entity);
+        return entity;
+    }
+
+    private TestingLabEntity getAtlEntityById(final Long entityId) throws EntityRetrievalException {
+        TestingLabEntity entity = null;
+
+        String queryStr = "SELECT atl from TestingLabEntity atl " + "LEFT OUTER JOIN FETCH atl.address "
+                + "WHERE (atl.id = :entityid)" + " AND (atl.deleted = false)";
+
+        Query query = entityManager.createQuery(queryStr, TestingLabEntity.class);
+        query.setParameter("entityid", entityId);
+        List<TestingLabEntity> result = query.getResultList();
+
+        if (result == null || result.size() == 0) {
+            String msg = errorMessageUtil.getMessage("atl.notFound");
+            throw new EntityRetrievalException(msg);
+        } else if (result.size() > 1) {
+            throw new EntityRetrievalException("Data error. Duplicate certificaiton body id in database.");
+        } else if (result.size() == 1) {
+            entity = result.get(0);
+        }
+        return entity;
+    }
+
+    private UserEntity getUserEntityById(final Long userId) throws EntityRetrievalException {
+
+        UserEntity user = null;
+
+        Query query = entityManager.createQuery(
+                "from UserEntity where (NOT deleted = true) " + "AND (user_id = :userid) ", UserEntity.class);
+        query.setParameter("userid", userId);
+        List<UserEntity> result = query.getResultList();
+
+        if (result == null || result.size() == 0) {
+            String msg = String.format(messageSource.getMessage(new DefaultMessageSourceResolvable("user.notFound"),
```

Comment:
Didn't notice this one before.. can you double check that errorMessageUtil is used everywhere in this class and the messageSource.getMessage thing is not used anywhere?